superset/connectors/base/models.py

```diff
@@ -185,6 +186,34 @@ def data(self):
             'verbose_map': verbose_map,
         }
 
+    @staticmethod
+    def filter_values_handler(
+            values, target_column_is_numeric=False, is_list_target=False):
+        def handle_single_value(v):
+            # backward compatibility with previous <select> components
+            if isinstance(v, basestring):
+                v = v.strip().strip("'").strip('"')
+                if target_column_is_numeric:
+                    # For backwards compatibility and edge cases
+                    # where a column data type might have changed
+                    v = utils.string_to_num(v)
+            if v == '<NULL>':
+                return None
+            elif v == '<empty string>':
+                return ''
+            return v
+        if isinstance(values, (list, tuple)):
+            values = [handle_single_value(v) for v in values]
+        values = handle_single_value(values)
```

Comment:
This is a bit confusing... I'd add the second call to an `else` block (unless I'm missing something):

```python
if isinstance(values, (list, tuple)):
    values = [handle_single_value(v) for v in values]
else:
    values = handle_single_value(values)
```